### PR TITLE
RPE-677: More unit tests for validator

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Validate zip file received from Exela](https://tools.hmcts.net/jira/browse/RPE-677)

### Change description ###

Including some contract breaking metafiles to test out our schema validation. Checking most cases so it can be caught and covered if rules are changed in the future

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+    @After
+    public void tearDown() {
+        capture.flush();
+    }
+
+    @Test
+    public void should_not_parse_envelope_with_no_scannable_items_in() {
+        tryToGetEnvelope(
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java]: I think these test cases are not easily readable. what do you think of below approach?

```@Test
    public void should_not_parse_envelope_with_no_scannable_items_in() throws Exception {
        //Given
        byte[] noScannableMetaFile = IOUtils.toByteArray(getClass().getResource("/metafiles/invalid/no-scannables.json"));

        //when
        AbstractThrowableAssert actualAssertion = assertThatCode(() -> validator.validate(noScannableMetaFile));

        //then
        actualAssertion.isInstanceOf(InvalidEnvelopeSchemaException.class);

        actualAssertion.hasMessage("Failed validation against schema:\n" +
            "\tobject has missing required properties ([\"scannable_items\"])");

    }
```

I know you have tried to extract common code into a method but that's making it a bit difficult to read.

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java]: i think it's a lot of repetitive work and will put developers to sleep while scrolling. also `AbstractThrowableAssert` has self type check which is kind-of weird and i didn't check how to correctly assign it as a variable.

that said, i can still convert everything how you suggest

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.InvalidEnvelopeSchemaException;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+public class MetafileJsonValidatorTestForInvalidFiles {
+
+    @Rule
+    public OutputCapture capture = new OutputCapture();
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java]: Are you using this capture anywhere ?

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    public void should_not_parse_envelope_with_invalid_zip_file_name_format() throws IOException {
+        // given
+        byte[] metafile = getMetafile("/metafiles/invalid/invalid-zip-file-name-format.json");
+
+        // when
+        AbstractThrowableAssert assertion = tryToValidateMetafile(metafile);
+
+        // then
+        assertion
+            .hasMessageStartingWith("Failed validation against schema:\n\tECMA 262 regex ")
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java]: is ECMA 262 regex added by Validator ?

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java]: yes

**Pull request merged**